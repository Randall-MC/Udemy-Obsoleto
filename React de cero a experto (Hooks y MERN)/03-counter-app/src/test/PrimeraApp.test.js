import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {
    
    test("Debe mostrar <PrimeraApp /> correctamente", () => {

        const saludo = 'Mi Primera App';
        const wrapper = shallow( <PrimeraApp saludo = { saludo }/> ); 
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrar subtitulo enviado por props', () => {

        const saludo = 'Mi primera App';
        const subTitulo = 'con React';
        const wrapper = shallow(
            <PrimeraApp
                saludo = { saludo }
                subtitulo = { subTitulo }
            />)
        const textoParrafo = wrapper.find('p').text();
        // console.log( textoParrafo );
        expect( textoParrafo ).toBe( subTitulo );

    });

    // test('Debe de mostrar el mensaje "Mi Primera App"', () => {
        
    //     const saludo = "Mi Primera App";
    //     const { getByText} = render( <PrimeraApp saludo = { saludo } />);
    //     expect( getByText( saludo )).toBeInTheDocument();

    // });
});