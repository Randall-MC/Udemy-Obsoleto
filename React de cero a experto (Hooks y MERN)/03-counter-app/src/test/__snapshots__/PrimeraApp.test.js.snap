// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Mi Primera App
  </h1>
  <p>
    con React
  </p>
</Fragment>
`;
